satellite/overlay/service.go

```diff
@@ -81,6 +81,9 @@ type DB interface {
 	GetGracefulExitIncompleteByTimeFrame(ctx context.Context, begin, end time.Time) (exitingNodes storj.NodeIDList, err error)
 	// GetExitStatus returns a node's graceful exit status.
 	GetExitStatus(ctx context.Context, nodeID storj.NodeID) (exitStatus *ExitStatus, err error)
+
+	// GetNodeIPs returns a list of IP addresses associated with given node IDs.
+	GetNodeIPs(ctx context.Context, nodeIDs []storj.NodeID) (nodeIPs []string, err error)
```

Comment:
Is there a reason why we can't pass in the `DistinctIP` preference into the `FindStorageNodes` call?

Reply:
the reason for this function is that we already have a list of nodes (from the pointer), and we want to exclude those nodes' IPs from our `FindStorageNodes` call. If we add the `DistinctIP flag` to the `FindStorageNodes` request, then the nodes we get back would have distinct IPs from each other, but some of those IPs could be the same as those of the `excludedNodes` that we pass in. IP duplicate checking isn't built into `SelectStorageNodes`.